src/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImplAsync.cs

```diff
@@ -3593,7 +3593,7 @@ private enum ParseTextFunction
                 ThrowInvalidChar(_ps.chars, _ps.charsUsed, _ps.charPos + offset);
             }
             //should never hit here
-            throw new Exception();
+            throw new XmlException("This program location is thought to be unreachable.");            
```

Comment:
found on src\System.Reflection.Metadata\src\System\Reflection\Internal\Utilities\ExceptionUtilities.cs i think XmlException is better to understand the "Source" of exception in case of hit.

Reply:
I think it might be a good idea to:
- try to reuse a string from resources
- if not string can be reused, add a new one

Reply:
done